File: itp/circle_in_rectangle/src/circle_in_rectangle.scala
/* ITP1_2_D: Circle in a Rectangle */

import scala.io.StdIn.{readLine, readInt}

object Main {
  val centerX: Int = 0
  val centerY: Int = 0

  def main(args: Array[String]): Unit = {
    val inputs = readLine().split(" ").map(_.toInt)
    val within = withinRect(inputs)
    println( if (within) "Yes" else "No" )
  }

  def withinRect(inputs: Seq[Int]): Boolean = {
    val Seq(width, height, x, y, radius) = inputs
    val distances = List(
      x - centerX,
      y - centerY,
      width - x,
      height - y
    )
    distances.forall(_ >= radius)
  }
}

File: itp/circle_in_rectangle/src/circle_in_rectangle.cpp
// 与えられる x, y, r が以下の条件を全て満たせば
// 円は長方形の中にある。
// 0 <= x <= W
// 0 <= y <= H
// x + r <= W && 0 <= x - r
// y + r <= H && 0 <= y - r
//
// r == 0 がありうる ..?


/* ITP1_2_D: Circle in a Rectangle */

#include <iostream>
using namespace std;

bool between(int from, int v, int to) {
    return from <= v && v <= to;
}

int main() {
    int W, H, x, y, r, c_x = 0, c_y = 0;
    cin >> W >> H >> x >> y >> r;

    bool is_inside = between(c_x + r, x, W - r) && between(c_y + r, y, H - r);
    cout << ( is_inside ? "Yes" : "No" ) << endl;
    return 0;
}
